File: Programmers/Level 2/스킬트리/Solution.py
from collections import deque

def solution(skill, skill_trees):
    answer = 0
    
    for tree in skill_trees:
        queue = deque(list(skill))
        for t in tree:
            if t in queue:
                if t != queue.popleft():
                    break
        else:
            answer += 1
    
    return answer

'''
정확성  테스트
테스트 1 〉	통과 (0.01ms, 10.2MB)
테스트 2 〉	통과 (0.01ms, 10.1MB)
테스트 3 〉	통과 (0.01ms, 10.3MB)
테스트 4 〉	통과 (0.01ms, 10.2MB)
테스트 5 〉	통과 (0.02ms, 10.2MB)
테스트 6 〉	통과 (0.01ms, 10.1MB)
테스트 7 〉	통과 (0.02ms, 10.1MB)
테스트 8 〉	통과 (0.02ms, 10.3MB)
테스트 9 〉	통과 (0.01ms, 10.2MB)
테스트 10 〉	통과 (0.02ms, 10.2MB)
테스트 11 〉	통과 (0.02ms, 10.3MB)
테스트 12 〉	통과 (0.02ms, 10.1MB)
테스트 13 〉	통과 (0.02ms, 10.1MB)
테스트 14 〉	통과 (0.02ms, 10.2MB)
'''

File: Programmers/Level 2/스킬트리/Solution.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(string skill, vector<string> skill_trees) {
    int answer = 0;

    for(int i=0;i<skill_trees.size();i++){
        int count = 0;
        bool check = true;
        for(int j=0;j<skill_trees[i].size();j++){
            if(skill.find(skill_trees[i][j])==string::npos)
                continue;
            else if(skill.find(skill_trees[i][j])==count)
                count++;
            else{
                check = false;
                break;
            }
        }

        if(check)
            answer++;

    }

    return answer;
}